Bond/Bond+Arrays.swift

```diff
@@ -806,6 +949,13 @@ public extension DynamicArray
   public func filter(f: T -> Bool) -> DynamicArray<T> {
     return _filter(self, f)
   }
+  
+  /**
+  Note: The returned flattened DynamicArray should not be modified directly.
+  */
+  static public func flatten(nestedDynamicArray: DynamicArray<DynamicArray<T>>) -> DynamicArray<T> {
```

Comment:
I'm not sure of a way to make flatten an instance function. Basically this only works on DynamicArrays of DynamicArrays. I went ahead and moved it to a module-level function though